include/seqan3/alignment/configuration/align_config_free_ends.hpp

```diff
@@ -0,0 +1,173 @@
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides the free ends gap configuration.
+ * \author Jörg Winkler <j.winkler AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <iostream>
+
+#include <seqan3/alignment/configuration/utility.hpp>
+#include <seqan3/alignment/scoring/gap_scheme.hpp>
+#include <seqan3/core/algorithm/all.hpp>
+#include <seqan3/core/detail/strong_type.hpp>
+#include <seqan3/core/concept/core_language.hpp>
+#include <seqan3/core/metafunction/basic.hpp>
+#include <seqan3/core/metafunction/template_inspection.hpp>
+
+namespace seqan3
+{
+
+/*!\brief Enum class for all supported free ends specifications that can be used with
+ * seqan3::detail::align_config_free_ends.
+ * \ingroup configuration
+ */
+enum struct free_ends_enum
+{
+    none       = 0,
+    seq1_front = 1 << 0,
+    seq1_back  = 1 << 1,
+    seq2_front = 1 << 2,
+    seq2_back  = 1 << 3,
+    seq1       = seq1_front | seq1_back,
+    seq2       = seq2_front | seq2_back,
+    front      = seq1_front | seq2_front,
+    back       = seq1_back | seq2_back,
+    all        = seq1 | seq2
+};
+
+//!\cond
+template <>
+constexpr bool add_enum_bitwise_operators<seqan3::free_ends_enum> = true;
+//!\endcond
+
+} // namespace seqan3
+
+namespace seqan3::detail
+{
+/*!\brief A configuration element for free end gaps.
+ * \ingroup configuration
+ * \tparam pos A compile time value for this configuration.
+ */
+template <free_ends_enum pos = free_ends_enum::none>
+struct align_config_free_ends
+{
+    //! \brief Holds the actual free ends settings.
+    free_ends_enum value = pos;
+};
+
+/*!\brief The free ends adaptor enabling pipe notation.
+ * \ingroup configuration
+ * \tparam free_ends_cfg_t A template-template class specifying the actual free ends settings.
+ */
+template <template <free_ends_enum> typename free_ends_cfg_t, free_ends_enum e>
+struct align_config_free_ends_adaptor : public configuration_fn_base<align_config_free_ends_adaptor<free_ends_cfg_t, e>>
+{
+    /*!\brief Adds to the configuration a free ends configuration element.
+     * \tparam configuration_type The type of the configuration to be extended.
+     * \param[in] cfg The configuration to be extended.
+     * \param[in] fe The value to be set.
+     * \returns A new configuration containing the free ends configuration element.
+     */
+    template <typename configuration_type>
+    //!\cond
+        requires is_algorithm_configuration_v<remove_cvref_t<configuration_type>>
+    //!\endcond
+    constexpr auto invoke(configuration_type && cfg, free_ends_enum const fe) const
+    {
+        static_assert(is_valid_alignment_configuration_v<align_cfg::id::free_ends, remove_cvref_t<configuration_type>>,
+                      SEQAN3_INVALID_CONFIG(align_cfg::id::free_ends));
+
+        return std::forward<configuration_type>(cfg).push_front(align_config_free_ends<>{std::move(fe)});
```

Comment:
You need for this the deferred config class. Otherwise, the value represented does not represent the template argument of the config. That could lead to misuse in the code.